common/lib/xmodule/xmodule/vertical_block.py

```diff
@@ -82,7 +83,15 @@ def _student_or_public_view(self, context, view):
 
         # pylint: disable=no-member
         for child in child_blocks:
-            if context.get('hide_access_error_blocks') and getattr(child, 'has_access_error', False):
+
```

Comment:
Minor: It might be useful to extract the business logic of determining whether a block should be displayed or not out into it's own function to make it easier to understand & test all the cases - e.g. `def _should_display_block`

Reply:
ooh I like that idea- it's different code now, but I'll use that idea to adjust my new implementation